core/chains/evm/gas/block_history_estimator.go

```diff
@@ -240,7 +240,7 @@ func (b *BlockHistoryEstimator) GetLegacyGas(_ context.Context, _ []byte, gasLim
 			"Using EvmGasPriceDefault as fallback.", "blocks", b.getBlockHistoryNumbers())
 		gasPrice = b.config.EvmGasPriceDefault()
 	}
-	gasPrice = capGasPrice(gasPrice, maxGasPriceWei, b.config)
+	gasPrice = capGasPrice(gasPrice, maxGasPriceWei, b.config.EvmMaxGasPriceWei())
```

Comment:
The idea is to inject in dependency rather than having to pass in one whole config just for an attribute